graphbolt/src/csc_sampling_graph.cc

```diff
@@ -636,11 +636,108 @@ inline int64_t NonUniformPick(
     return num_positive_probs;
   } else {
     if (!replace) fanout = std::min(fanout, num_positive_probs);
-    std::memcpy(
-        picked_data_ptr,
-        (torch::multinomial(local_probs, fanout, replace) + offset)
-            .data_ptr<PickedType>(),
-        fanout * sizeof(PickedType));
+    if (fanout == 0) return 0;
+    TORCH_CHECK(
+        ((local_probs.max() < INFINITY) & (local_probs.min() >= 0))
+            .item()
+            .to<bool>(),
+        "Invalid probs_or_mask (contains either `inf`, `nan` or element < 0).");
+    TORCH_CHECK(
+        !(local_probs.sum() == 0).item().to<bool>(),
+        "Invalid probs_or_mask (sum of probabilities <= 0).");
```

Comment:
In previous check, there is local_probs.min() >= 0. Why do we need to check !local_probs.sum() == 0?